aerleon/lib/nsxt.py

```diff
@@ -0,0 +1,392 @@
+            return []
+
+
+class Term(aclgenerator.Term):
+    """Creates a single ACL Term for NSX-T."""
+
+    def __init__(self, term, filter_type, applied_to=None, af=4):
+        self.term = term
+        # Our caller should have already verified the address family.
+        assert af in (4, 6)
+        self.af = af
+        self.filter_type = filter_type
+        self.applied_to = applied_to
+
+    def __str__(self):
+        """Convert term to a rule string.
+
+        Returns:
+          A rule as a string.
+
+        """
+        # Verify platform specific terms. Skip whole term if platform does not
+        # match.
+        if self.term.platform:
+            if 'nsxt' not in self.term.platform:
+                return ''
+        if self.term.platform_exclude:
+            if 'nsxt' in self.term.platform_exclude:
+                return ''
+
+        term_keywords = self.term.__dict__
+        unsupported_keywords = []
+        for key in term_keywords:
+            if term_keywords[key]:
+                if ('translated' not in key) and (key not in _NSXT_SUPPORTED_KEYWORDS):
+                    unsupported_keywords.append(key)
+        if unsupported_keywords:
+            logging.warning(
+                'WARNING: The keywords %s in Term %s are not supported ' 'in Nsxt ',
+                unsupported_keywords,
+                self.term.name,
+            )
+
+        name = self.term.name
+
+        notes = ''
+        if self.term.comment:
+            notes = '\n'.join(self.term.comment)
+
+        action = 'ALLOW'
+        if self.term.action:
+            action = _ACTION_TABLE.get(self.term.action[0])
+
+        source_address = ['ANY']
+        if self.term.source_address:
+            source_address = []
+            for i in self.term.source_address:
+                source_address.append(str(i))
+
+        destination_address = ['ANY']
+        if self.term.destination_address:
+            destination_address = []
+            for i in self.term.destination_address:
+                destination_address.append(str(i))
+
+        rule = {
+            'action': action,
+            'resource_type': 'Rule',
+            'display_name': name,
+            'source_groups': source_address,
+            'destination_groups': destination_address,
+            # Set mandatory services to ANY, as service_entries will be used
+            'services': ['ANY'],
+            'profiles': ['ANY'],
+            'scope': ['ANY'],
+            'logged': bool(self.term.logging),
+            'notes': notes,
+            'direction': 'IN_OUT',
+            'ip_protocol': 'IPV4_IPV6',
+        }
+
+        if self.term.protocol:
+            icmp_types = []
+            services = []
+            if self.term.icmp_type:
+                icmp_types = self.NormalizeIcmpTypes(
+                    self.term.icmp_type, self.term.protocol, self.af
+                )
+
+            protocol = [self.PROTO_MAP.get(p) for p in self.term.protocol]
```

Comment:
`term.protocol` can contain a number (e.g. `protocol:: 1` instead of `protocol:: icmp`). In those cases, `self.PROTO_MAP.get(p)` will return None.

Does NSX-T allow numeric protocols in its output? If not there is a mapping, PROTO_MAP_BY_NUMBER, that can map back from number to name.